ts/components/session/SessionNicknameDialog.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import { ConversationController } from '../../session/conversations/ConversationController';
+import { SessionModal } from './SessionModal';
+import { SessionButton, SessionButtonColor } from './SessionButton';
+import { SessionHtmlRenderer } from './SessionHTMLRenderer';
+import { SessionIcon, SessionIconSize, SessionIconType } from './icon';
+import { DefaultTheme, withTheme } from 'styled-components';
+
+type Props = {
+  message: string;
+  messageSub: string;
+  title: string;
+  placeholder?: string;
+  onOk?: any;
+  onClose?: any;
+  onClickOk: any;
+  onClickClose: any;
+  okText?: string;
+  cancelText?: string;
+  hideCancel: boolean;
+  okTheme: SessionButtonColor;
+  closeTheme: SessionButtonColor;
+  sessionIcon?: SessionIconType;
+  iconSize?: SessionIconSize;
+  theme: DefaultTheme;
+  convoId?: string;
+};
+
+const SessionNicknameInner = (props: Props) => {
+  const {
+    title = '',
+    message,
+    messageSub = '',
+    okTheme = SessionButtonColor.Primary,
+    closeTheme = SessionButtonColor.Primary,
+    onClickOk,
+    onClickClose,
+    hideCancel = false,
+    sessionIcon,
+    iconSize,
+    convoId,
+    placeholder,
+  } = props;
+
+  const okText = props.okText || window.i18n('ok');
+  const cancelText = props.cancelText || window.i18n('cancel');
+  const showHeader = !!props.title;
```

Comment:
probably no need to keep that dynamic. Not sure if the header is useful for this dialog, but just set it to true or false.